smt.py

```diff
@@ -306,18 +369,99 @@ def merge_smtlib(it: Iterator[SMTLIB]) -> SMTLIB:
     return SMTLIB('\n'.join(it))
 
 
+raw_mem_acc_prelude = ['''
+(define-fun load-word8 ((m {MemSort}) (p (_ BitVec 64)))
+	(_ BitVec 8)
+	(select m p)
+)
+''',
+                       '''
+(define-fun load-word16 ((m {MemSort}) (p (_ BitVec 64)))
+	(_ BitVec 16)
+	(concat (select m (bvadd p #x0000000000000001))
+		(select m p)
+	)
+)
+''',
+                       '''
+(define-fun load-word32 ((m {MemSort}) (p (_ BitVec 64)))
+	(_ BitVec 32)
+	(concat
+		(concat (select m (bvadd p #x0000000000000003))
+				(select m (bvadd p #x0000000000000002)))
+		(concat (select m (bvadd p #x0000000000000001))
+				(select m p))
+	)
+)
+''',
+                       '''
+(define-fun load-word64 ((m {MemSort}) (p (_ BitVec 64)))
+	(_ BitVec 64)
+	(concat (load-word32 m (bvadd p #x0000000000000004))
+			(load-word32 m p)
+	)
+)
+''',
+                       '''
+(define-fun store-word8 ((m {MemSort}) (p (_ BitVec 64)) (v (_ BitVec 8)))
+	{MemSort}
+	(store m p v)
+)
+''',
+                       '''
+(define-fun store-word16 ((m {MemSort}) (p (_ BitVec 64)) (v (_ BitVec 16)))
+	{MemSort}
+	(store-word8
+		(store-word8 m p ((_ extract 7 0) v))
+		(bvadd p #x0000000000000001)
+		((_ extract 15 8) v)
+	)
+)
+''',
+                       '''
+(define-fun store-word32 ((m {MemSort}) (p (_ BitVec 64)) (v (_ BitVec 32)))
+	{MemSort}
+	(store-word16
+		(store-word16 m p ((_ extract 15 0) v))
+		(bvadd p #x0000000000000002)
+		((_ extract 31 16) v)
+	)
+)
+''',
+                       '''
+(define-fun store-word64 ((m {MemSort}) (p (_ BitVec 64)) (v (_ BitVec 64)))
+	{MemSort}
+	(store-word32
+		(store-word32 m p ((_ extract 31 0) v))
+		(bvadd p #x0000000000000004)
+		((_ extract 63 32) v)
+	)
+)''']
```

Comment:
i'm willing to trust this prelude is correct, though I'd feel better if we had some test for it, something like in `gen.py`

Also I think it's a bit misleading to make them generic in `MemSort`, the primitive function definitions only work for on type of Mem, namely the one we use.